llvm/test/CodeGen/RISCV/rvv/fixed-vectors-insert.ll

```diff
@@ -65,9 +65,8 @@ define <32 x i32> @insertelt_v32i32_31(<32 x i32> %a, i32 %y) {
 ; CHECK-LABEL: insertelt_v32i32_31:
 ; CHECK:       # %bb.0:
 ; CHECK-NEXT:    li a1, 32
-; CHECK-NEXT:    vsetvli zero, a1, e32, m1, ta, ma
-; CHECK-NEXT:    vmv.s.x v16, a0
 ; CHECK-NEXT:    vsetvli zero, a1, e32, m8, ta, ma
```

Comment:
This is an interesting sub-case as we shouldn't need to be reasoning about ratios here.  Why didn't the existing logic get this case given only LMUL is changing?  I think this might just be the lack of LI handling in isNonZeroAVL?

Reply:
Good point, I'm not quite sure why this part wasn't already caught since it should just be an entire VTYPE substitution.
Note to self though, this is one of the cases the forward-pass-ratio-mutation in #69788 won't catch because it's the first instruction in the BB